e2e_tests/tests/per_provider/normal_tests/export_key.rs

```diff
@@ -0,0 +1,211 @@
+        return Ok(());
+    }
+
+    let key_name = String::from("check_export_possible");
+
+    let key_attributes = Attributes {
+        lifetime: Lifetime::Persistent,
+        key_type: Type::RsaKeyPair,
+        bits: 1024,
+        policy: Policy {
+            usage_flags: UsageFlags {
+                sign_hash: false,
+                verify_hash: false,
+                sign_message: false,
+                verify_message: false,
+                export: true,
+                encrypt: false,
+                decrypt: false,
+                cache: false,
+                copy: false,
+                derive: false,
+            },
+            permitted_algorithms: Algorithm::AsymmetricSignature(
+                AsymmetricSignature::RsaPkcs1v15Sign {
+                    hash_alg: Hash::Sha256.into(),
+                },
+            ),
+        },
+    };
+
+    client.generate_key(key_name.clone(), key_attributes)?;
+
+    let _public_key = client.export_key(key_name)?;
+
+    Ok(())
+}
+
+
+#[test]
+fn check_export_not_possible() {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return;
+    }
+
+    let key_name = String::from("check_export_not_possible");
+
+    let key_attributes = Attributes {
+        lifetime: Lifetime::Persistent,
+        key_type: Type::RsaKeyPair,
+        bits: 1024,
+        policy: Policy {
+            usage_flags: UsageFlags {
+                sign_hash: false,
+                verify_hash: false,
+                sign_message: false,
+                verify_message: false,
+                export: false,
+                encrypt: false,
+                decrypt: false,
+                cache: false,
+                copy: false,
+                derive: false,
+            },
+            permitted_algorithms: Algorithm::AsymmetricSignature(
+                AsymmetricSignature::RsaPkcs1v15Sign {
+                    hash_alg: Hash::Sha256.into(),
+                },
+            ),
+        },
+    };
+
+    let _generated_key = client.generate_key(key_name.clone(), key_attributes).unwrap();
+
+    let _exported_key = client.export_key(key_name).unwrap_err();
+}
+
+#[test]
+fn export_ecc_key() {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return;
+    }
+
+    let key_name = String::from("export_ecc_key");
+    let _generated_key = client.generate_ecc_key_pair(key_name.clone()).unwrap();
+    let _exported_key = client.export_key(key_name).unwrap();
+}
+
+#[test]
+fn export_key_matches_import() {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return;
+    }
+
+    let key_name = String::from("export_key_matches_import");
+
+    let decoded_key = base64::decode(PRIVATE_KEY).unwrap();
+    client
+        .import_rsa_key_pair(key_name.clone(), decoded_key.to_vec())
```

Comment:
I don't remember seeing when this was added, so apologies for the late comment: I think it's worth including in the method name what the key will be usable for - I thought it would be for signing, initially

Reply:
Ah, it says in the method comment what it is for (same as `import_rsa_public_key`) but I agree it should be added to the method name 🙂.